src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -274,18 +274,21 @@ ov::npuw::LLMCompiledModel::LLMCompiledModel(const std::shared_ptr<ov::Model>& m
     reshape_to_static(kvcache_model, 1u, m_kvcache_desc.total_size, axes);
 
     auto npudesc = extract_npu_descriptor(plugin);
-
     ov::AnyMap properties_copy = std::move(other_props);
-    auto prefill_config = get_default_prefill_config(model, npudesc);
-    // NB: GENERATE_HINT is only applicable for default generate config!
+
+    auto prefill_config = opt_or_default(prefill_config_opt, get_default_prefill_config(prefill_model, npudesc));
+
     const ::intel_npu::npuw::llm::GenerateHint generate_hint = m_cfg.get<::intel_npu::NPUW_LLM_GENERATE_HINT>();
     LOG_DEBUG("9. Passed GENERATE_HINT: " << std::string(::intel_npu::NPUW_LLM_GENERATE_HINT::toString(generate_hint)));
-    auto generate_config = get_default_generate_config(model, npudesc, generate_hint);
+    // NB: GENERATE_HINT is only applicable for default generate config!
+    if (generate_config_opt.has_value() && npuw_llm_props.count(ov::intel_npu::npuw::llm::generate_hint.name())) {
+        OPENVINO_THROW("GENERATE_HINT is only applicable for default generate config!");
+    }
+    auto generate_config =
+        opt_or_default(generate_config_opt, get_default_generate_config(model, npudesc, generate_hint));
+
     merge_config_with(prefill_config, properties_copy);
     merge_config_with(generate_config, properties_copy);
-    // FIXME: Drop CACHE_DIR option if NPUW is enabled
```

Comment:
Why is it dropped?

Reply:
Because it should be handled on the GenAI side, here we already passed through the NPU plugin, that chooses us (`npuw::LLMCompiledModel`) and checked `CACHE_DIR` existance

Reply:
Given this config, will `NPU` plugin handle `CACHE_DIR`? Or it will be responsibility of `NPUW`?
```
USE_NPUW: YES,
NPUW_LLM_PIPELINE: YES,
CACHE_DIR: "..."
```

Reply:
Yes, `NPU` plugin handles it here: https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_npu/src/plugin/src/plugin.cpp#L637

Reply:
Got it, thanks!